File: leetcode/self_dividing_numbers/self_dividing_numbers.kt
class Solution {
  
    fun isSelfDividing(num: Int): Boolean {
      val input_number = num
      var result = 0
      input_number.toString().toCharArray().toList()
    	.map { it.toString().toInt() }
    	.filter {it != 0}
        .forEach {
        if (input_number % it.toInt() == 0) {
          result += 1
        } 
      }
      return result == input_number.toString().length
}
  
    fun selfDividingNumbers(left: Int, right: Int): List<Int> {
        var return_list = mutableListOf<Int>()
        (left..right).forEach {
          if (isSelfDividing(it)) {
            return_list.add(it)
          }
        }
      return return_list
    }
}

File: leetcode/self_dividing_numbers/self_dividing_numbers.py
class Solution(object):
    def isSelfDivisible(self, number): 
        return_number = 0
        for digit in str(number):
          try:
            if int(number) % int(digit) == 0:
              return_number += 1
          except ZeroDivisionError:
              pass
        return return_number == len(str(number))
      
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        return_list = []
        for number in range(left, right+1):
          if self.isSelfDivisible(number):
            return_list.append(number)
        return return_list
